### DISCUSSION TITLE
ArtCNN example script

### DISCUSSION THREAD

**Discussion by ghost** [AmusementClub/vs-mlrt#101] [Q&A]
I want to use it on mpv can you provide example vs script?

**Comment by WolframRhodium** [comment]
ref: https://github.com/hooke007/MPV_lazy/discussions/471
what's the current issue?

```python3
import vapoursynth as vs
import vsmlrt

clip = video_in.fmtc.bitdepth(bits=32)
clip_y = core.std.ShufflePlanes(clip, 0, vs.GRAY)
clip_y = vsmlrt.ArtCNN(clip_y, model=2, backend=vsmlrt.Backend.ORT_DML(fp16=True))
clip_uv = core.resize.Bicubic(clip, clip_y.width, clip_y.height)
clip = core.std.ShufflePlanes([clip_y, clip_uv], [0, 1, 2], vs.YUV)
clip = core.fmtc.bitdepth(clip, bits=8)
clip.set_output()
```

**Reply by ghost** [reply]
@WolframRhodium I got https://0.0g.gg/?dbfdbdc4b135f32d#3iAVopgmbuHv1qWNeX43uo4ecrvfBbUHDGhCrmLU1oYm while trying hooke007's implementation. Also can't I use trt instead of ort?

**Reply by WolframRhodium** [reply]
The error is expected since it is not an RGB model but a grayscale model. trt can be used, but the chroma variants require the latest plugins in GitHub actions.
